src/multiprocess/multiprocess_memory_limit.c

```
@@ -1367,21 +1414,87 @@ int update_host_pid() {
+                break;
+            }
         }
     }
-    if (!found) {
+
+    if (slot == NULL) {
         LOG_ERROR("HOST PID NOT FOUND. %d",hostpid);
         return -1;
     }
+
+    LOG_INFO("SET PID= %d", hostpid);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)